packages/js/product-editor/src/components/variations-table/variation-actions-menus/variation-actions.tsx

```diff
@@ -0,0 +1,150 @@
+
+/**
+ * Internal dependencies
+ */
+import { VariationActionsMenuProps } from './types';
+import { TRACKS_SOURCE } from '../../../constants';
+import { ShippingMenuItem } from '../shipping-menu-item';
+import { InventoryMenuItem } from '../inventory-menu-item';
+import { PricingMenuItem } from '../pricing-menu-item';
+import { ToggleVisibilityMenuItem } from '../toggle-visibility-menu-item';
+import { DownloadsMenuItem } from '../downloads-menu-item';
+import { VariationActionsMenuItem } from './variation-actions-menu-item';
+import { QUICK_UPDATE, SINGLE_UPDATE } from './constants';
+import { UpdateStockMenuItem } from '../update-stock-menu-item';
+import { SetListPriceMenuItem } from '../set-list-price-menu-item';
+
+export function VariationActions( {
+	selection,
+	onChange,
+	onDelete,
+	onClose,
+	type = SINGLE_UPDATE,
+}: VariationActionsMenuProps & {
+	onClose: () => void;
+	type: string;
+} ) {
+	const isQuickUpdate = type === QUICK_UPDATE;
+	return (
+		<div
+			className={ classNames( {
+				'components-dropdown-menu__menu': isQuickUpdate,
+			} ) }
+		>
+			<MenuGroup
+				label={
+					isQuickUpdate
+						? undefined
+						: sprintf(
+								/** Translators: Variation ID */
+								__( 'Variation Id: %s', 'woocommerce' ),
+								( selection as ProductVariation ).id
+						  )
+				}
+			>
+				{ isQuickUpdate ? (
+					<>
+						<UpdateStockMenuItem
+							selection={ selection }
+							onChange={ onChange }
+							onClose={ onClose }
+						/>
+						<SetListPriceMenuItem
+							selection={ selection }
+							onChange={ onChange }
+							onClose={ onClose }
+						/>
+					</>
+				) : (
+					<MenuItem
+						href={ ( selection as ProductVariation ).permalink }
+						target="_blank"
+						rel="noreferrer"
+						onClick={ () => {
+							recordEvent( 'product_variations_preview', {
+								source: TRACKS_SOURCE,
+								variation_id: ( selection as ProductVariation )
+									.id,
+							} );
+						} }
+					>
+						{ __( 'Preview', 'woocommerce' ) }
+					</MenuItem>
+				) }
+
+				<ToggleVisibilityMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+				/>
+				<VariationActionsMenuItem.Slot
+					group={ '_main' }
+					type={ type }
+				/>
+			</MenuGroup>
+			<MenuGroup>
+				<PricingMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+					type={ type }
+				/>
+				<InventoryMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+					type={ type }
+				/>
+				<ShippingMenuItem
+					selection={ selection }
+					onChange={ onChange }
+					onClose={ onClose }
+					type={ type }
+				/>
+				{ window.wcAdminFeatures[ 'product-virtual-downloadable' ] && (
+					<DownloadsMenuItem
+						selection={ selection }
+						onChange={ onChange }
+						onClose={ onClose }
+						type={ type }
+					/>
+				) }
+				<VariationActionsMenuItem.Slot
+					group={ '_secondary' }
```

Comment:
I think we may not want to allow a slot at this level (or, at least not document/support it). The name of the group might be better as `top-level`.

We should check with @j111q to see where extensions should put menu items that don't fall under any of the existing groups, and whether they should be able to create their own groups.

Reply:
Good call! I'll rename those groups initially to `top-level`, `secondary`, and `tertiary` and wait for Jill's answer to confirm if we should keep or remove those slots. I like the idea of providing a lot of flexibility, but as you said, maybe that would be too much.